pkg/apis/serving/v1alpha1/domainmapping_types.go

```diff
@@ -70,6 +70,14 @@ type DomainMappingList struct {
 	Items []DomainMapping `json:"items"`
 }
 
+// SecretTLS wrapper for TLS SecretName.
+type SecretTLS struct {
+	// SecretName is the name of the existing secret used to terminate TLS traffic.
+	SecretName string `json:"secretName"`
+	// SecretNamespace is the namespace of the existing secret used to terminate SSL traffic. If empty the namespace of the DomainMapping is assumed.
+	SecretNamespace string `json:"secretNamespace"`
```

Comment:
Is it alright to allow users to specify `SecretNamespace`?
I am worried that it causes a vulnerability as they can steal a secret in any other namespaces.

Reply:
Do we do something similar in [ReconcileCertificate](https://github.com/knative/serving/blob/main/pkg/reconciler/accessor/networking/certificate.go#L42) if the certificate exists ?

Caveat: I don't have much context in this area so I maybe quite wrong :)

Reply:
> Is it alright to allow users to specify SecretNamespace?
I am worried that it causes a vulnerability as they can use a secret in any other namespaces.

This is a good point and it would depend on each `net-*` implementation to do the right thing.

I believe contour requires delegation - thus an operator would need to be involved.

Unsure about other ingresses

Reply:
we have [some validation in KReference](https://github.com/knative/pkg/blob/main/apis/duck/v1/knative_reference.go#L71) (which is what domainmapping.spec.ref is obviously) to check the namespace is the same as the parent object, maybe we should add something similar for this.

Reply:
Sourcing some info here:

https://knative.slack.com/archives/CA9RHBGJX/p1622035644007800

I'm thinking for now we remove `SecretNamespace` and we can always add it back later if there's a demand for it